internal/core/server/handler.go

```diff
@@ -21,8 +25,26 @@ type Config struct {
 	TaskTracker *task.TaskTracker
 }
 
-func BuildRootMux(prefix string, cfg *Config) *http.ServeMux {
+func BuildRootMux(prefix string, cfg *Config, tokenClient token.TokenClient) *http.ServeMux {
+	authMiddleware := func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			token := strings.TrimPrefix(req.Header.Get("Authorization"), "Bearer ")
+			cookie, err := req.Cookie("token")
+			if token == "" && err == nil {
+				token = cookie.Value
+			}
+
+			authed := tokenClient.CheckToken(token)
+			if !authed {
+				httputil.WriteError(w, req, errutil.NewHTTPError(http.StatusUnauthorized, "Bad or no token"))
```

Comment:
nit: error messages typically are not complete sentences and therefore should not be capitalized.

Reply:
So I wasn't sure how best to do this for messages that are going to show up directly in the UI. There it looks much better capitalised and possibly punctuated.

Reply:
Ah, good point. We need to think through our error strategy more completely. What we probably want is some stable error code that we can potentially use for localization someday.